<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Mainwindow</class>
 <widget class="QWidget" name="Mainwindow">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1177</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SG_SPHSImulation</string>
  </property>
  <widget class="SPHSimulation" name="openGLWidget">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>600</height>
    </rect>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>890</x>
     <y>560</y>
     <width>171</width>
     <height>31</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="QuitButtonLayout">
    <item>
     <widget class="QPushButton" name="quitButton">
      <property name="text">
       <string>Quit</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>1040</x>
     <y>70</y>
     <width>131</width>
     <height>31</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="RestartSimButtonLayout">
    <item>
     <widget class="QPushButton" name="restartButton">
      <property name="text">
       <string>Restart</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>10</y>
     <width>181</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Courier</family>
     <pointsize>12</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>SG SPH SIMULATION</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>150</y>
     <width>115</width>
     <height>22</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QLineEdit" name="v_numParticles">
      <property name="text">
       <string>1000</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_3">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>180</y>
     <width>115</width>
     <height>22</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QLineEdit" name="v_mass">
      <property name="text">
       <string>1.0</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_4">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>210</y>
     <width>115</width>
     <height>22</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <widget class="QLineEdit" name="v_radius">
      <property name="text">
       <string>0.15</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_5">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>149</y>
     <width>110</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <item>
     <widget class="QLabel" name="numParticles">
      <property name="font">
       <font>
        <family>DejaVu Serif</family>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>num particles</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_6">
   <property name="geometry">
    <rect>
     <x>1040</x>
     <y>30</y>
     <width>131</width>
     <height>31</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="RestartSimButtonLayout_2">
    <item>
     <widget class="QPushButton" name="pauseButton">
      <property name="text">
       <string>Pause</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_7">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>180</y>
     <width>110</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_5">
    <item>
     <widget class="QLabel" name="mass">
      <property name="font">
       <font>
        <family>DejaVu Serif</family>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>mass</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_8">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>210</y>
     <width>110</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_6">
    <item>
     <widget class="QLabel" name="radius">
      <property name="font">
       <font>
        <family>DejaVu Serif</family>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>radius</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>800</x>
     <y>100</y>
     <width>371</width>
     <height>20</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="num_part_4">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>120</y>
     <width>108</width>
     <height>19</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>DejaVu Sans</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>Particle Controls</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="Line" name="line_2">
   <property name="geometry">
    <rect>
     <x>800</x>
     <y>260</y>
     <width>321</width>
     <height>20</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="num_part_5">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>280</y>
     <width>108</width>
     <height>19</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>DejaVu Sans</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>Gravity</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_9">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>240</y>
     <width>110</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_7">
    <item>
     <widget class="QLabel" name="density">
      <property name="font">
       <font>
        <family>DejaVu Serif</family>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>density</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_10">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>240</y>
     <width>115</width>
     <height>22</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_8">
    <item>
     <widget class="QLineEdit" name="v_density">
      <property name="text">
       <string>25</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_11">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>310</y>
     <width>110</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_9">
    <item>
     <widget class="QLabel" name="gravity">
      <property name="font">
       <font>
        <family>DejaVu Serif</family>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Gravity</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_12">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>310</y>
     <width>115</width>
     <height>22</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_10">
    <item>
     <widget class="QLineEdit" name="v_gravity">
      <property name="text">
       <string>-9.8</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_13">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>370</y>
     <width>119</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_11">
    <item>
     <widget class="QLabel" name="pConstant">
      <property name="font">
       <font>
        <family>DejaVu Serif</family>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Pressure Const</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_14">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>370</y>
     <width>115</width>
     <height>22</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_12">
    <item>
     <widget class="QLineEdit" name="v_pConst">
      <property name="text">
       <string>0.05</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_15">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>400</y>
     <width>110</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_13">
    <item>
     <widget class="QLabel" name="p_gamma">
      <property name="font">
       <font>
        <family>DejaVu Serif</family>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Gamma</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_16">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>400</y>
     <width>115</width>
     <height>22</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_14">
    <item>
     <widget class="QLineEdit" name="v_pGamma">
      <property name="text">
       <string>3</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="num_part_10">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>340</y>
     <width>108</width>
     <height>19</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>DejaVu Sans</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>Pressure</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="num_part_11">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>430</y>
     <width>108</width>
     <height>19</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>DejaVu Sans</family>
     <pointsize>10</pointsize>
     <weight>50</weight>
     <bold>false</bold>
    </font>
   </property>
   <property name="text">
    <string>Viscocity</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_17">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>460</y>
     <width>118</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_15">
    <item>
     <widget class="QLabel" name="vConstant">
      <property name="font">
       <font>
        <family>DejaVu Serif</family>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Viscosity Constant</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_18">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>460</y>
     <width>115</width>
     <height>22</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_16">
    <item>
     <widget class="QLineEdit" name="v_vConst">
      <property name="text">
       <string>3</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_19">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>490</y>
     <width>118</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_17">
    <item>
     <widget class="QLabel" name="vE">
      <property name="font">
       <font>
        <family>DejaVu Serif</family>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Episilon</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_20">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>490</y>
     <width>115</width>
     <height>22</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_18">
    <item>
     <widget class="QLineEdit" name="v_vE">
      <property name="text">
       <string>0.15</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_21">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>520</y>
     <width>118</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_19">
    <item>
     <widget class="QLabel" name="WallSticky">
      <property name="font">
       <font>
        <family>DejaVu Serif</family>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Wall Sticky</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_22">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>520</y>
     <width>115</width>
     <height>22</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_20">
    <item>
     <widget class="QLineEdit" name="v_wallSticky">
      <property name="text">
       <string>0.5</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_23">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>80</y>
     <width>110</width>
     <height>21</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_21">
    <item>
     <widget class="QLabel" name="numParticles_2">
      <property name="font">
       <font>
        <family>DejaVu Serif</family>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>timeStep</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLineEdit" name="v_timeStep">
   <property name="geometry">
    <rect>
     <x>940</x>
     <y>80</y>
     <width>71</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>0.005</string>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>SPHSimulation</class>
   <extends>QOpenGLWidget</extends>
   <header>sphsimulation.h</header>
   <slots>
    <slot>settimestep(QString)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <slots>
  <signal>timestepChanged(QString)</signal>
  <slot>winQuit()</slot>
  <slot>settimestep()</slot>
 </slots>
</ui>
